DQM/HGCal/plugins/HGCalTriggerClient.cc

```diff
@@ -57,24 +65,136 @@ class HGCalTriggerClient : public DQMEDAnalyzer {
   std::map<uint32_t, HGCalSiCellChannelInfo> eleidtoSiInfo_;
 
   const bool debug_;
-  const unsigned prescale_;
+  const int
+      verbose_;  //level 5 (all), 4(frequent erros), 3(errors), 2(rare errors), 1(very rare errors/error under study)
+  unsigned runNumber_;
+  const unsigned prescale_;  // set to 1 to fill histogram for every event
   const unsigned min_num_evts_;
   unsigned num_processed_ = 0;
 };
 
 HGCalTriggerClient::HGCalTriggerClient(const edm::ParameterSet& iConfig)
     : trigRawToken_(consumes<FEDRawDataCollection>(iConfig.getParameter<edm::InputTag>("RawTrigData"))),
       metadataToken_(consumes<HGCalTestSystemMetaData>(iConfig.getParameter<edm::InputTag>("MetaData"))),
+      //recRun_(consumes<hgcal::SlinkFromRaw>(iConfig.getParameter<edm::InputTag>("RecRun"))),
       moduleInfoToken_(
           esConsumes<HGCalCondSerializableModuleInfo, HGCalCondSerializableModuleInfoRcd, edm::Transition::BeginRun>()),
       siModuleInfoToken_(esConsumes<HGCalCondSerializableSiCellChannelInfo,
                                     HGCalCondSerializableSiCellChannelInfoRcd,
                                     edm::Transition::BeginRun>()),
       debug_(iConfig.getUntrackedParameter<bool>("debug")),
+      verbose_(iConfig.getUntrackedParameter<int>("verbose")),
+      runNumber_(iConfig.getUntrackedParameter<unsigned>("runNumber")),
       prescale_(std::max(1u, iConfig.getParameter<unsigned>("Prescale"))),
       min_num_evts_(iConfig.getParameter<unsigned>("MinimumNumEvents")) {}
 
-HGCalTriggerClient::~HGCalTriggerClient() { LogDebug("HGCalTriggerClient") << "End of the job" << std::endl; }
+HGCalTriggerClient::~HGCalTriggerClient() {
+  LogDebug("HGCalTriggerClient") << "End of the job"
+                                 << "\n";
+}
+
+// print 64-bit words of event
+void HGCalTriggerClient::eventDump(const uint64_t* payload,
+                                   const size_t payload_size,
+                                   edm::LogSystem& out,
+                                   int verbose = 5) {
+  if (verbose_ >= verbose) {
+    for (unsigned i = 0; i < payload_size; i++) {
+      out << std::setw(4) << i << "  ";
+      out << std::hex << std::setfill('0');
+      out << std::setw(16) << payload[i] << '\n';
+      out << std::dec << std::setfill(' ');
+    }
+  }
+  out << "\n";
+}
+
+//Checks of matches with 0xFECAFE
+bool HGCalTriggerClient::is0xFECAFE(const uint32_t word) {
+  if (((word >> 8) & 0xFFFFFF) == 16698110)
```

Comment:
So:
- By construction, `word>>8` is identical to `word>>8 & 0xFFFFFF`.
- `0xFECAFE` is not even used in the code?

I suggest `if( (word>>8) == 0xFECAFE )` or `if( word & 0xFFFFFF00 == 0xFECAFE00)`.

Reply:
Also, the `if` is not needed, e.g. `return ( word & 0xFFFFFF00 == 0xFECAFE00 )`.

Reply:
Necessary modifications have been applied (see [8165d07](https://github.com/CMS-HGCAL/cmssw/pull/86/commits/8165d077af54e5ab2fcceb6c538b92669bbe113d))
However, kept the if structure.